src/Giveaway.js

```diff
@@ -123,10 +124,33 @@ class Giveaway extends EventEmitter {
          */
         this.allowedMentions = options.allowedMentions;
         /**
-         * The giveaway data.
-         * @type {GiveawayData}
+         * The buttons of the giveaway, if any.
+         * @type {?ButtonsObject}
          */
-        this.options = options;
+        this.buttons = options.buttons?.join ? options.buttons : null;
+        /**
+         * The entrant ids for this giveaway, if buttons are used.
+         * @type {?Discord.Snowflake[]}
+         */
+        this.entrantIds = options.entrantIds ?? this.buttons ? [] : null;
+        /**
+         * Giveaway options which need to be processed in a getter or function.
+         * @type {Object}
+         * @property {Discord.EmojiIdentifierResolvable} [reaction] The reaction to participate in the giveaway.
+         * @property {boolean} [botsCanWin] If bots can win the giveaway.
+         * @property {Discord.PermissionResolvable[]} [exemptPermissions] Members with any of these permissions will not be able to win the giveaway.
+         * @property {string} [exemptMembers] Filter function to exempt members from winning the giveaway.
+         * @property {string} [bonusEntries] The array of BonusEntry objects for the giveaway.
+         * @property {Discord.ColorResolvable} [embedColor] The color of the giveaway embed when it is running.
+         * @property {Discord.ColorResolvable} [embedColorEnd] The color of the giveaway embed when it has ended.
+         * @property {LastChanceOptions} [lastChance] The options for the last chance system.
+         * @property {PauseOptions} [pauseOptions] The options for the pause system.
+         * @property {boolean} [isDrop] If the giveaway is a drop, or not.<br>Drop means that if the amount of valid entrants to the giveaway is the same as "winnerCount" then it immediately ends.
+         */
+        this.options = Object.keys(options).reduce((obj, key) => {
```

Comment:
Did this cause otherwise the entrant array would be in there and also the other options which are already a standalone `this.`. --> saves memory I guess XD